src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,218 @@
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in carsDB.", e);
+        }
+        deleteDriverList(car);
+        createDriverList(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.id AS car_id, model, manufacturer_id, "
+                + "m.name as manufacturer_name, m.country "
+                + "FROM cars AS c "
+                + "INNER JOIN manufacturers AS m ON m.id = c.manufacturer_id "
+                + "INNER JOIN cars_drivers AS cd ON cd.car_id = c.id "
+                + "INNER JOIN drivers as d ON cd.driver_id = d.id "
+                + "WHERE cd.driver_id = ? AND c.is_deleted = FALSE AND d.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car's list by driver id"
+                    + driverId, e);
+        }
+        for (Car car: cars) {
+            car.setDrivers(getDriversByCarID(car.getId()));
+        }
+        return cars;
+    }
+
+    private void createDriverList(Car car) {
+        if (car.getDrivers().isEmpty()) {
+            return;
+        }
+        StringBuilder values = new StringBuilder();
+        for (var ignored : car.getDrivers()) {
+            values.append((values.length() == 0) ? "" : ",").append("(?,?)");
+        }
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUES "
+                + values;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            int counter = 1;
+            for (Driver driver : car.getDrivers()) {
+                statement.setLong(counter++, car.getId());
+                statement.setLong(counter++, driver.getId());
+            }
+            statement.executeUpdate();
```

Comment:
With such change it should work properly without StringBuilder usage:
```suggestion
            statement.setLong(1, car.getId());
            for (Driver driver : car.getDrivers()) {
                statement.setLong(2, driver.getId());
                statement.executeUpdate();
            }
```

Reply:
in such case if a car has 10 drivers we`d execute 10 queries to DB instead of one

Reply:
its ok